src/bindings/python/flux/util.py

```diff
@@ -596,3 +606,114 @@ def render(
             max_level=level,
             truncate=limit,
         )
+
+
+def get_searchpath(var, suffix=""):
+    value = os.getenv(var)
+    if value is None:
+        return []
+    return [f"{x}{suffix}" for x in value.split(":")]
+
+
+#  Slightly modified from https://stackoverflow.com/posts/7205107/timeline
+def dict_merge(src, new, path=None):
+    "merges dict new into dict src"
+    if path is None:
+        path = []
+    for key in new:
+        if key in src:
+            if isinstance(src[key], dict) and isinstance(new[key], dict):
+                dict_merge(src[key], new[key], path + [str(key)])
+            elif src[key] == new[key]:
+                pass  # same leaf value
+            else:
+                #  Default is to override:
+                src[key] = new[key]
+        else:
+            src[key] = new[key]
+    return src
```

Comment:
for your implementation, I don't think `path` is needed.  In the original it appears it was just used for debugging.